packages/valory/contracts/gnosis_safe/tests/test_contract.py

```diff
@@ -628,3 +685,224 @@ def test_verify_reverted(self, *_: Any) -> None:
+        )
+        assert verified["verified"], f"Not verified: {verified}"
+        time.sleep(1)
+        remove_events = self.contract.get_removed_owner_events(
+            ledger_api=self.ledger_api,
+            contract_address=self.contract_address,
+            removed_owner=owner_to_be_removed,
+            from_block=prev_block,
+        ).get("data")
+
+        assert remove_events is not None, "a RemovedOwner event was expected"
+        assert len(remove_events) == 1, "1 RemovedOwner event was expected"
+        assert (
+            remove_events[0].get("owner") == owner_to_be_removed
+        ), "a different owner than expected was removed"
+
+    def test_swap(self) -> None:  # pylint: disable=too-many-locals
+        """Test owner swapping."""
+        assert self.contract_address is not None
+        old_owner = self.owners()[1]
+        new_owner = KEY_PAIRS[-1][0]
+        value = 0
+        data_str = self.contract.get_swap_owner_data(
+            ledger_api=self.ledger_api,
+            contract_address=self.contract_address,
+            old_owner=old_owner,
+            new_owner=new_owner,
+        ).get("data")
+        data = bytes.fromhex(data_str[2:])  # strip 0x before converting to bytes
+
+        sender = crypto_registry.make(
+            EthereumCrypto.identifier, private_key_path=ETHEREUM_KEY_PATH_1
+        )
+        assert sender.address == self.owners()[1]
+        receiver = crypto_registry.make(
+            EthereumCrypto.identifier, private_key_path=ETHEREUM_KEY_PATH_2
+        )
+        assert receiver.address == self.owners()[2]
+        fourth = crypto_registry.make(
+            EthereumCrypto.identifier, private_key_path=ETHEREUM_KEY_PATH_3
+        )
+        assert fourth.address == self.owners()[3]
+        cryptos = [self.deployer_crypto, sender, receiver, fourth]
+        tx_hash = self.contract.get_raw_safe_transaction_hash(
+            ledger_api=self.ledger_api,
+            contract_address=self.contract_address,
+            to_address=self.contract_address,
+            value=value,
+            data=data,
+        )["tx_hash"]
+        b_tx_hash = binascii.unhexlify(cast(str, tx_hash)[2:])
+        signatures_by_owners = {
+            crypto.address: crypto.sign_message(b_tx_hash, is_deprecated_mode=True)[2:]
+            for crypto in cryptos
+        }
+        assert list(signatures_by_owners.keys()) == self.owners()
+
+        tx = self.contract.get_raw_safe_transaction(
+            ledger_api=self.ledger_api,
+            contract_address=self.contract_address,
+            sender_address=sender.address,
+            owners=(self.deployer_crypto.address.lower(),),
+            to_address=self.contract_address,
+            value=value,
+            data=data,
+            signatures_by_owner={
+                self.deployer_crypto.address.lower(): signatures_by_owners[
+                    self.deployer_crypto.address
+                ]
+            },
+        )
```

Comment:
this is exact same code block as [here](https://github.com/valory-xyz/open-autonomy/pull/1469/files#diff-23ee345d2f557f9dd1cd468e759f8cfecb17f49ee813d84105dc1d3826029cdcR710-R749). Please refactor